package/MDAnalysis/analysis/hydrogenbonds/wbridge_analysis.py

```diff
@@ -1341,7 +1341,8 @@ def traverse_water_network(graph, node, end, route, maxdepth, result):
             else:
                 if node in end:
                     # check if any duplication happens
-                    if len(route) == len(set(route)):
+                    heavy_atom = [line[3] or line[2] for line in route]
+                    if len(heavy_atom) == len(set(heavy_atom)):
                         add_route(result, route)
                 else:
                     for new_node in graph[node]:
```

Comment:
Sorry, my earlier comment wasn't very well worded - what I meant to say is that, at the moment, you only run the check for duplicates once a route has reached `end`; if you moved it down a little so you instead checked before adding each new node, you could abort duplicates then and save time spent on building up routes that will later be discarded. 
I'd assume that'd overall work out faster at higher orders, though I'm less sure for lower orders (which presumably most people will be using)?

Reply:
I see your point now.
So the major bottleneck is the distance search and the network building is almost instant. So either way, it is not gonna change the speed it runs. 

As you have said, in the case of the higher-order water network and duplications happen very often, checking the duplication at each iteration will give better performance as one could terminate the search early. But I guess we would then need to carry another hash table around for the checking procedure.

Reply:
Ok, thanks - that makes sense. It should be fine as is, then :)